fcrepo-persistence-ocfl/src/main/java/org/fcrepo/persistence/ocfl/OCFLPersistentStorageSession.java

```diff
@@ -0,0 +1,132 @@
+import org.fcrepo.kernel.api.exception.ItemNotFoundException;
+import org.fcrepo.kernel.api.exception.RepositoryRuntimeException;
+import org.fcrepo.kernel.api.models.FedoraResource;
+import org.fcrepo.kernel.api.models.Tombstone;
+import org.fcrepo.persistence.api.PersistentStorageSession;
+import org.fcrepo.persistence.api.exceptions.PersistentStorageException;
+
+/**
+ * OCFL Persistent Storage class.
+ *
+ * @author whikloj
+ * @since 2019-09-20
+ */
+public class OCFLPersistentStorageSession implements PersistentStorageSession {
+
+    /**
+     * Fedora transaction id for the session.
+     */
+    private final String sessionId;
+
+    /**
+     * Constructor
+     *
+     * @param txId Fedora transaction id.
+     */
+    protected OCFLPersistentStorageSession(final String txId) {
+        this.sessionId = txId;
+    }
+
+    /**
+     * Constructor
+     */
+    protected OCFLPersistentStorageSession() {
+        this.sessionId = null;
+    }
+
+    @Override
+    public String getTxId() {
+        return this.sessionId;
+    }
+
+    @Override
+    public FedoraResource create(final FedoraResource resource) throws CannotCreateResourceException {
+        actionNeedsWrite();
+        return null;
+    }
+
+    @Override
+    public FedoraResource update(final FedoraResource resource) throws CannotCreateResourceException {
+        actionNeedsWrite();
+        return null;
+    }
+
+    @Override
+    public Tombstone delete(final FedoraResource resource) throws RepositoryRuntimeException {
+        actionNeedsWrite();
+        return null;
+    }
+
+    @Override
+    public FedoraResource read(final String identifier) throws ItemNotFoundException {
+        actionNeedsWrite();
+        return null;
+    }
+
+    @Override
+    public FedoraResource read(final String identifier, final Instant version) throws ItemNotFoundException {
+        if (!isReadOnly()) {
+            throw new PersistentStorageException("Can't read versions in a session");
+        }
+        return null;
+    }
+
+    @Override
+    public void commit() throws RepositoryRuntimeException {
+        if (isReadOnly()) {
+            // No changes to commit.
+            return;
+        }
+        // commit changes.
+    }
+
+    @Override
+    public void rollback() throws RepositoryRuntimeException {
+        if (isReadOnly()) {
+            // No changes to rollback
+            return;
+        }
+        // rollback changes
+    }
+
+    /**
+     * Check if we are in a read-only session.
+     *
+     * @return whether we are read-only (ie. no transaction).
+     */
+    private boolean isReadOnly() {
+        return this.sessionId == null;
+    }
+
+    /**
+     * Utility to throw exception if trying to perform write operation on read-only session.
+     */
+    private void actionNeedsWrite() {
+        if (isReadOnly()) {
+            throw new PersistentStorageException("Session is read-only");
```

Comment:
UnsupportedOperationException here as well?

Reply:
Sorry do you want a different exception to be the result here?

Reply:
I was thinking using the UnsupportedOperationException, but if PersistentStorageException is checked (as we agreed above) then PersistentStorageException makes sense.